rust/sdk/examples/chat/cmdline.rs

```diff
@@ -12,40 +12,115 @@ pub enum Cmdline {
     Reconnect(ActoRef<Messages>),
 }
 
-fn mk_text_area() -> TextArea<'static> {
+fn mk_message_text_area() -> TextArea<'static> {
     let mut text_area = TextArea::default();
     text_area.set_block(
         Block::default()
-            .title("your message (Enter to send, Ctrl-C to quit)")
+            .title("your message (Enter to send, Ctrl-I to Edit Identity, Ctrl-C to quit)")
             .borders(Borders::ALL),
     );
     text_area
 }
 
+fn mk_identity_text_area(s: &String) -> TextArea<'static> {
+    let mut text_area: TextArea<'_> = TextArea::default();
+    text_area.set_block(
+        Block::default()
+            .title("your identity (Enter to apply, Escape to cancel or use default, Ctrl-C to quit)")
+            .borders(Borders::ALL),
+    );
+    text_area.insert_str(s);
+    text_area
+}
+
+pub struct Identity {
+    pub edit: Option<TextArea<'static>>,
+    pub val: TextArea<'static>,
+}
+
+impl Default for Identity {
+    fn default() -> Self {
+        let me: String = "me".into();
+        Self {
+            edit: Some(mk_identity_text_area(&me)),
+            val: mk_identity_text_area(&me),
+        }
+    }
+}
+
 pub async fn cmdline(
     mut cell: ActoCell<Cmdline, impl ActoRuntime>,
     display: ActoRef<Display>,
     mut messages: ActoRef<Messages>,
 ) {
-    let text_area = Writer::new(mk_text_area());
+    let identity = Writer::new(Identity::default());
+    let text_area = Writer::new(mk_message_text_area());
+    display.send(Display::UpdateIdentity(identity.reader()));
     display.send(Display::Cmdline(text_area.reader()));
 
     while let ActoInput::Message(msg) = cell.recv().await {
         match msg {
             Cmdline::Event(event) => {
-                if let Event::Key(key) = event {
-                    // FIXME modifiers are not recognised, so no ctrl-enter
-                    if key.code == KeyCode::Enter {
-                        let mut text_area = text_area.write();
-                        let text = std::mem::replace(&mut *text_area, mk_text_area())
-                            .into_lines()
-                            .join("\n");
-                        tracing::info!("publishing message: {}", text);
-                        messages.send(Messages::Publish(messages::Event::new("me".to_owned(), text)));
-                    } else {
-                        text_area.write().input(key);
+                // Prevent double input because of KeyEventKind::Release
+                let key = match event {
+                    Event::Key(key) if key.kind == KeyEventKind::Press || key.kind == KeyEventKind::Repeat => key,
+                    _ => continue,
+                };
+
+                let is_editing_identity = { (identity.read()).edit.clone() };
+                match is_editing_identity {
```

Comment:
I’d replace the match with

```rust
let edit_name = identity.read().edit.is_some();
if edit_name { ...
```

Reply:
Keeping this because it is linked to this https://github.com/Actyx/Actyx/pull/554#discussion_r1360634705